src/services/search-builder.js

```diff
@@ -148,15 +148,22 @@ function SearchBuilder(model, opts, params, fieldNamesRequested) {
           pushCondition(condition, field.field);
         }
       } else if (field.type === 'Number') {
-        let value = Number(params.search);
-
-        if (!Number.isNaN(value)) {
-          if (Number.isInteger(value) && !Number.isSafeInteger(value)) {
-            // NOTE: Numbers higher than MAX_SAFE_INTEGER need to be handled as
-            //       strings to circumvent precision problems
+        const searchAsNumber = Number(params.search);
+        let value;
+
+        if (!Number.isNaN(searchAsNumber)) {
+          if (Number.isInteger(searchAsNumber)
+            && !Number.isSafeInteger(searchAsNumber)
+            && model.rawAttributes[field.field].type instanceof DataTypes.BIGINT) {
+            // Numbers higher than MAX_SAFE_INTEGER need to be handled as strings to circumvent
+            // precision problems only if the field type is a big int.
             value = params.search;
+          } else {
+            value = searchAsNumber;
           }
+        }
 
+        if (value !== undefined) {
```

Comment:
why is this check needed 🤔

Reply:
It's because, in some cases, we don't want to add a condition.
For instance, the case that interests us here, if you search for a number that is not a safeInteger (too big) and the field on which to search is not of type big int (let's say it's a regular integer). You don't want to add a condition on it otherwise Postgres will try to compare a regular integer with a big int value and this makes the crash `... is out of range for type integer`.
Is it clear?

Reply:
Crystal